src/backend/be_utils.js

```diff
@@ -149,39 +149,41 @@ export async function be_utils_get_client_name(cliente_id) {
 }
 
 export async function be_utils_get_dashboard_data() {
-    let clients = await wixData.query(BD_CLIENTE).find({suppressAuth: true});
-    let transactions = await wixData.query(BD_TRANSACOES).eq("situacao", "aprovada").find({suppressAuth: true});
+    let clients_query = wixData.aggregate(BD_CLIENTE)
+    let transactions_query = wixData.aggregate(BD_TRANSACOES)
 
-    let clients_total = clients.length;
+    let filter = wixData.filter().eq("situacao", "aprovada");
+    let filter_cashback_used = wixData.filter().eq("situacao", "aprovada").eq("tipo", "pagamento_saldo");
 
-    let total_abastecido_app =  transactions.items.reduce((total, item) => {
-                                    return total + item.valor;
-                                }, 0);
+    let clients_count = (await clients_query
+                        .count()
+                        .run()).items[0].count;
 
-    let average_abastecimento = (total_abastecido_app/100)/transactions.length;
+    let total_abastecido_app = (await transactions_query
+                        .filter(filter)
+                        .sum("valor")
+                        .run()).items[0].valorSum;
 
-    let cashback_used = transactions.items.reduce((total, item) => {
-                            if (item.tipo == "pagamento_saldo")
-                                return total + item.valorTipo;
-                            return total;
-                        }, 0);
+    let average_abastecimento = (await transactions_query
+                        .filter(filter)
+                        .avg("valor")
+                        .run()).items[0].valorAvg / 100;
 
-    let cashback_to_be_used =   clients.items.reduce((total, item) => {
-                                    return total + item.saldo;
-                                }, 0);
+    let cashback_used = (await transactions_query
+                        .filter(filter_cashback_used)
+                        .sum("valorTipo")
+                        .run()).items[0].valorTipoSum * -1;
 
-    let total_paid =    transactions.items.reduce((total, item) => {
-                            if (item.tipo == "cashback") // if is casback the total paid is the total of the transaction
-                                return total + item.valor;
-                            if (item.valor + item.valorTipo == 0) // if is not 'cashback', check if the transaction is paid with ALL 'saldo'
-                                return total;
-                            return total + (item.valor + item.valorTipo); // if is not 'cashback' and the transaction is not paid with ALL 'saldo', the total paid is the total of the transaction minus the 'saldo' used (that is already negative)
-                        }, 0);
+    let cashback_to_be_used = (await clients_query
+                        .sum("saldo")
+                        .run()).items[0].saldoSum;
+
```

Comment:
it seems that almost all these queries could be done in parallel, and you can wait for the promises with await Promises.all

Reply:
Nice tip, I added for this queries.